core/src/main/java/org/infinispan/util/logging/Log.java

```diff
@@ -2002,4 +2002,7 @@ TimeoutException coordinatorTimeoutWaitingForView(int expectedViewId, int curren
 
    @Message(value = "A store cannot be configured with both preload and purgeOnStartup", id = 589)
    CacheConfigurationException preloadAndPurgeOnStartupConflict();
+
+   @Message(value = "Store must specify a location when global state is disabled", id = 590)
+   CacheConfigurationException storeLocationRequired();
```

Comment:
To allow for better validation of other store location attributes, we should create an overloaded version of this message.

```java
   @Message(value = "Store must specify the '%s' attribute when global state is disabled", id = 590)
   CacheConfigurationException storeLocationRequired(String attributeName);
```